profitbricks/client.py

```diff
@@ -1736,30 +1719,30 @@ def _perform_request(self, url, type='GET',
         # print('Headers: '+str(headers))
         # print('########################################')
 
-        if type == 'POST':
-            response = self._perform_post(url, data, headers)
-        elif type == 'PUT':
-            response = self._perform_put(url, data, headers)
-        elif type == 'PATCH':
-            response = self._perform_patch(url, data, headers)
-        elif type == 'DELETE':
-            response = self._perform_delete(url, data, headers)
-            if response.status_code == 202:
-                return True
-        elif type == 'POST-ACTION-JSON':
-            response = self._perform_post_action_json(url, data, headers)
-            if response.status_code == 202:
-                return response
-            elif response.status_code == 401:
-                raise response.raise_for_status()
-        elif type == 'POST-ACTION':
-            response = self._perform_post_action(url, data, headers)
+        if type == 'POST' or type == 'PUT':
```

Comment:
One could also imagine the content types and request verb being stored in a hash/dictionary indexed by the types and use that hash in the _wrapped_request method.